client/src/services/informationManager.ts

```diff
@@ -0,0 +1,155 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { MsAlertModal, MsAlertModalConfig, MsReportTheme } from '@/components/core';
+import { ToastManager } from '@/services/toastManager';
+import { modalController } from '@ionic/vue';
+import { v4 as uuid4 } from 'uuid';
+import { NotificationManager } from '@/services/notificationManager';
+
+export const InformationKey = 'information';
+
+export enum InformationLevel {
+  Info = 'INFO',
+  Success = 'SUCCESS',
+  Warning = 'WARNING',
+  Error = 'ERROR',
+}
+
+export interface IInformation {
+  message: string;
+  level: InformationLevel;
+  data?: object
+}
+
+export class Information {
+  id: string;
+  message: string;
+  level: InformationLevel;
+  data?: object;
+  constructor({ message, level, data = {} }: IInformation) {
+    this.id = uuid4();
+    this.message = message;
+    this.level = level;
+    this.data = data;
+  }
+
+  get theme(): MsReportTheme {
+    switch (this.level) {
+      case InformationLevel.Info:
+        return MsReportTheme.Info;
+      case InformationLevel.Success:
+        return MsReportTheme.Success;
+      case InformationLevel.Warning:
+        return MsReportTheme.Warning;
+      case InformationLevel.Error:
+        return MsReportTheme.Error;
+    }
+  }
+}
+
+export interface InformationOptions {
+  trace?: boolean;
+}
+
+export enum InformationPresentationMode {
+  Toast,
+  Modal,
+  Notification,
+  ToastAndNotification,
```

Comment:
Bitmasks are perfect to avoid this.

Reply:
I see the point after our last discussion about bitmasking, but I don't see anything we would want to "avoid" here.
It's clear, simple and control which cases we want to accept.
Can you detail more what's the matter here?

Reply:
Bitmask would make defining `ToastAndNotification` useless, and would facilitate the `switch` that checks this value.

```typescript
enum PresentationMode {
  // Using bit shifting because it makes it more confusing
  // and you look more like you know what you're doing, but
  // 1, 2 and 4 are acceptable values.
  Toast = 1 << 0,
  Modal = 1 << 1,
  Notification = 1 << 2,
}

function show(mode: number) {
  if (mode & PresentationMode.Toast) {
    showToast();
  }
  if (mode & PresentationMode.Modal) {
    showModal();
  }
  if (mode & PresentationMode.Notification) {
    addToNotification();
  }
}

show(PresentationMode.Toast | PresentationMode.Notification);
```
It makes it way easier to add or remove modes. For example, we currently have a trace option to print in the console, but it could become a mode too:

```typescript
enum PresentationMode {
  ...
  Console = 1 << 3,
}

function show(mode: number) {
  ...
  if (mode & PresentationMode.Console) {
    console.log(...);
  }
}

show(PresentationMode.Toast | PresentationMode.Modal | PresentationMode.Console);
```

Reply:
Hmm yh seems good to me, will replace.